Azure-Samples/azure-search-openai-demo
# Chat response

## Purpose

Implement the Chat Request part of the Chat App protocol:
https://github.com/Azure/azureml_run_specification/pull/1066/
That means the requests send "messages" and "context:overrides", instead of "history" and "overrides" as they sent before.

I elected to adhere to the protocol on multiple levels: in the TSX models, in the HTTP requests to the server, and in the run() method on Approach class. That should mean a high degree of compatibility with both other frontends as well as frameworks that use the protocol.
 
Also adjusts the response to return "context" instead of "extra_args" (a simple rename).

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* All tests still pass.
* But we could wait for my other PR to merge that adds way more e2e tests, to make sure they all pass, given this is a frontend-heavy change.
